### PR TITLE
[rand-token] Add typing for rand-token

### PR DESCRIPTION
**Pull request by azerella**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Doesn't look like this package has any types associate with it, let's fix it! I wasn't sure if `static generator(..)` and the `generator(...)` functions were the best way to handle the API design but it seems to be correct.

### CODE REVIEW

**Comment by sehrope** [comment]
I'm the maintainer rand-token. Please close this out as we'll be adding the types directly to the the module instead.

**Comment by adamzerella** [comment]
> I'm the maintainer rand-token. Please close this out as we'll be adding the types directly to the the module instead.

Added in https://github.com/sehrope/node-rand-token/pull/15 :+1:

File: types/rand-token/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type SourceType = 'default' | 'math' | 'crypto';
+
+interface GeneratorOptions {
+    source: SourceType;
+    chars: string;
+}
+
+declare class RandToken {
```
Comment (DanielRosenwasser) [inline types/rand-token/index.d.ts]: This isn't constructable, so don't make it a class. Just create an interface

```ts
export interface RandToken {
    generate(size: number, chars?: string[]): string;
}
```

and export each function at the top level

```ts
export function generator(generator: GeneratorOptions): RandToken;
export function uid(size: number): string;
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/rand-token/index.d.ts (hunk 1 above)
Comment (adamzerella) [reply types/rand-token/index.d.ts]: Cheers @DanielRosenwasser! I believe @sehrope wanted these types in the module repo itself, I'll create a PR on that repo updating them - I'll close this PR.

Good pickup with the constructable, not sure why I thought it was :confused:

**Pull request closed without merging**